[Edit] SQL: DATEADD() (#6823)
Codecademy/docs: content/sql/concepts/dates/terms/dateadd/dateadd.md

- `number`: The value to add to the specified interval. This can be positive to add time or negative to subtract time.
- `date`: The date value to which the interval will be added. This can be a date literal, a column containing date values, or a date returned by another function.

**Return value:**

The `DATEADD()` function returns a value of the same data type as the input `date` expression.

## Example 1: Basic Usage of the `DATEADD()` function

This example demonstrates how to add different time intervals to a date using the `DATEADD()` function:

```sql
-- Add 1 year to a date
SELECT DATEADD(year, 1, '2023-05-21') AS OneYearLater;

-- Add 3 months to a date
SELECT DATEADD(month, 3, '2023-05-21') AS ThreeMonthsLater;

-- Add 14 days to a date
SELECT DATEADD(day, 14, '2023-05-21') AS TwoWeeksLater;

-- Subtract 6 hours from a datetime
SELECT DATEADD(hour, -6, '2023-05-21 12:00:00') AS SixHoursEarlier;
```

The output of this code will be:

| OneYearLater | ThreeMonthsLater | TwoWeeksLater | SixHoursEarlier         |
| ------------ | ---------------- | ------------- | ----------------------- |
| 2024-05-21   | 2023-08-21       | 2023-06-04    | 2023-05-21 06:00:00.000 |

> This example shows how `DATEADD()` can perform a variety of date calculations by simply changing the interval parameter and the number value. Note that negative numbers subtract from the date rather than adding to it.

Reviewer: ```suggestion
This example shows how `DATEADD()` can perform a variety of date calculations by simply changing the interval parameter and the number value. 

>**Note:**  Negative numbers decrease the date rather than increase it.
```

## Example 3: Working with Variable Intervals

This example demonstrates how to use `DATEADD()` with variables to create flexible date calculations:

```sql
-- Declare variables for different parts of the calculation
DECLARE @StartDate DATETIME = '2023-01-15 08:30:00';
DECLARE @MonthsToAdd INT = 6;
DECLARE @DaysToAdd INT = 10;
DECLARE @HoursToAdd INT = 3;

-- Calculate the result date by applying multiple DATEADD() operations
SELECT
  @StartDate AS StartDate,
  DATEADD(month, @MonthsToAdd, @StartDate) AS AfterAddingMonths,
  DATEADD(day, @DaysToAdd,
    DATEADD(month, @MonthsToAdd, @StartDate)) AS AfterAddingMonthsAndDays,
  DATEADD(hour, @HoursToAdd,
    DATEADD(day, @DaysToAdd,
      DATEADD(month, @MonthsToAdd, @StartDate))) AS FinalDateTime;
```

The output generated by this code will be:

| StartDate               | AfterAddingMonths       | AfterAddingMonthsAndDays | FinalDateTime           |
| ----------------------- | ----------------------- | ------------------------ | ----------------------- |
| 2023-01-15 08:30:00.000 | 2023-07-15 08:30:00.000 | 2023-07-25 08:30:00.000  | 2023-07-25 11:30:00.000 |

> This example shows how DATEADD() can be nested and used with variables to create complex date calculations. This approach is particularly useful in stored procedures or dynamic SQL where the amount to add may vary based on different parameters or business rules.

Reviewer: ```suggestion
This example shows how `DATEADD()` can be nested and used with variables to create complex date calculations. This approach is particularly useful in stored procedures or dynamic SQL where the amount to add may vary based on different parameters or business rules.
```
